client.go

```diff
@@ -336,9 +332,7 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 
 	defer func() {
 		if netConn != nil {
-			if err := netConn.Close(); err != nil {
-				log.Printf("websocket: failed to close network connection: %v", err)
-			}
+			err = netConn.Close()
```

Comment:
It looks like your intent is to modify the function return value.   That's not what happens here.   [Demonstration](https://go.dev/play/p/5KnGoJG6LqZ).  

If the code is modified so that the return value is set, then primary error is ignored. [Demonstration](https://go.dev/play/p/3UAvH8ga0F4). 

Fix by removing this deferred function call  and changing the error returns below to 

      return nil, nil, errors.Join(err, netConn.Close())

 Delete the the `netConn = nil` statement at the end of the function.